crates/client-api/src/auth.rs

```diff
@@ -37,95 +38,91 @@ impl authorization::Credentials for SpacetimeCreds {
+        let Self { creds, identity } = self;
+        (
+            TypedHeader(SpacetimeIdentity(identity)),
+            TypedHeader(SpacetimeIdentityToken(creds)),
+        )
+    }
 }
 
-#[derive(Deserialize)]
-pub struct TokenQueryParam {
-    token: String,
+pub struct SpacetimeAuthHeader {
+    pub auth: Option<SpacetimeAuth>,
 }
 
 #[async_trait::async_trait]
 impl<S: NodeDelegate + Send + Sync> axum::extract::FromRequestParts<S> for SpacetimeAuthHeader {
     type Rejection = AuthorizationRejection;
     async fn from_request_parts(parts: &mut request::Parts, state: &S) -> Result<Self, Self::Rejection> {
-        match (
-            TypedHeader::from_request_parts(parts, state).await,
-            Query::<TokenQueryParam>::from_request_parts(parts, state).await,
-        ) {
-            (Ok(TypedHeader(headers::Authorization(creds @ SpacetimeCreds { .. }))), _) => {
-                let claims = creds
-                    .decode_token(state.public_key())
-                    .map_err(|e| AuthorizationRejection {
-                        reason: AuthorizationRejectionReason::Jwt(e.into_kind()),
-                    })?;
-                let auth = SpacetimeAuth {
-                    creds,
-                    identity: claims.identity,
-                };
-                Ok(Self { auth: Some(auth) })
-            }
-            (_, Ok(Query(query))) => {
-                let header =
-                    HeaderValue::from_str(&format!("Basic {}", query.token)).map_err(|_| AuthorizationRejection {
-                        reason: AuthorizationRejectionReason::MalformedTokenQueryString,
-                    })?;
-                let creds = SpacetimeCreds(authorization::Basic::decode(&header).ok_or(AuthorizationRejection {
-                    reason: AuthorizationRejectionReason::CantDecodeAuthorizationToken,
-                })?);
-                let claims = creds
-                    .decode_token(state.public_key())
-                    .map_err(|e| AuthorizationRejection {
-                        reason: AuthorizationRejectionReason::Jwt(e.into_kind()),
-                    })?;
-                let auth = SpacetimeAuth {
-                    creds,
-                    identity: claims.identity,
-                };
-                Ok(Self { auth: Some(auth) })
-            }
-            (Err(e), Err(_)) => match e.reason() {
-                // Leave it to handlers to decide on unauthorized requests.
-                TypedHeaderRejectionReason::Missing => Ok(Self { auth: None }),
-                _ => Err(AuthorizationRejection {
-                    reason: AuthorizationRejectionReason::Header(e),
-                }),
-            },
-        }
+        let Some(creds) = SpacetimeCreds::from_request_parts(parts)? else {
+            return Ok(Self { auth: None });
+        };
+        let claims = creds.decode_token(state.public_key())?;
```

Comment:
[more discussion on this refactor](https://github.com/clockworklabs/SpacetimeDB/pull/735/files?diff=unified&w=1#r1473233665)